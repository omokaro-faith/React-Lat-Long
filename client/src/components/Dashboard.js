import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import Loader from 'react-loader-spinner';
import GoogleApiWrapper from './MapWrapper';
import { getUsers }  from '../actions/users';

class Dashboard extends Component {
  state = {
    showingInfoWindow: false,
    activeMarker: {},
    selectedPlace: {},
    user: {
      company: {},
      address: {}
      }
  };

  onMouseoverMarker = (props, marker) => {
    const { users } = this.props;
    const { position } = props;
    users.filter((user) => {
      if(position === user.address.geo) {
        this.setState({
          selectedPlace: props,
          activeMarker: marker,
          showingInfoWindow: true,
          user
        });
      }
    });
  }

  componentDidMount(props, state) {
    this.props.getUsers();
  }

  onMouseoutMarker = () => {
    const { showingInfoWindow } = this.state;
    if (showingInfoWindow) {
      this.setState({
        showingInfoWindow: false,
        activeMarker: null,
        selectedPlace: {},
        user: {
          company: {},
          address: {}
        }
      });
    }
  }

  render() {
    const { 
      showingInfoWindow,
      activeMarker,
      selectedPlace,
      user
    } = this.state;
    const {
      users
    } = this.props;
    return (
      <GoogleApiWrapper
        showingInfoWindow={showingInfoWindow}
        activeMarker={activeMarker}
        selectedPlace={selectedPlace}
        onMouseoverMarker={this.onMouseoverMarker}
        users={users}
        onMouseoutMarker={this.onMouseoutMarker}
        perUserDetails={user}
      />
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  getUsers: () => dispatch(getUsers())
});

const mapStateToProps = (state) => ({
  users: state.users.users
})

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
