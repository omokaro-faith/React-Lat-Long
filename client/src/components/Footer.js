import React from 'react';

const Footer = () => (
  <div>
    <h2>Page Not Found</h2>
    <h3>We couldn't find what you were looking for.</h3>
  </div>
)

export default Footer;