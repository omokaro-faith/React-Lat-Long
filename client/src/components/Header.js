import React, { Component } from 'react';
import { NavLink, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export class Header extends Component {

  render() {
    const { startLogout, displayName, photoURL } = this.props;
    return (
      <nav className="navbar is-fixed-top is-light" id="navbarCustom" role="navigation" aria-label="main navigation">
        <div className="navbar-brand">
          <NavLink className="navbar-item" id="navbarCustom__logo" to="/dashboard" exact={true}>latLang</NavLink>
        </div>
        { displayName &&  <div className="navbar-end">
          <img id="navbarCustom__avatar" src={ photoURL }/>
          <button  id="navbarCustom__logout__btn" className="is-medium" onClick={startLogout} activeclassname="is-active"> Logout </button>
          <div className="navbar-item">
          </div>
        </div> }
      </nav>
    )
  }
}


const mapStateToProps = (state) => ({
  displayName: state.auth.displayName,
  photoURL: state.auth.photoURL
});

export default connect(mapStateToProps, {startLogout})(Header);