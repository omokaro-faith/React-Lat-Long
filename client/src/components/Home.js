import React from 'react';

const Home = () => (
  <p>This is the Home page</p>
)

export default Home;