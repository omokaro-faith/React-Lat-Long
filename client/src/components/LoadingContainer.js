import React from 'react';
import Loader from 'react-loader-spinner';
const LoadingContainer = (props) => (
  <div className="loader-item">
    <Loader
      type="ThreeDots"
      color="#6497b1"
      height="100"	
      width="100"
    />
  </div>
)

export default LoadingContainer;