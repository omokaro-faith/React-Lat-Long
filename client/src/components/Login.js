import React, { Component } from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';
import LoginImage from '../../images/btn-picture.jpg';

export class Login extends Component {
  constructor(props) {
    super(props);
  }

  Login = () => {
    const { startLogin } = this.props;
    startLogin();
  }

  render() {
    return(
    <div className="container">
      <div>
        <img id="container__image" src={LoginImage} width="100%" height="100%"/>
        <button className="button is-medium" onClick={ this.Login }><span>Login With Google</span></button>
      </div>
    </div>
    )
  }
}

export default connect(null, { 
  startLogin
 })(Login);
