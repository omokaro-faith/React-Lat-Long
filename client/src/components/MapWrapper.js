import React, { Fragment } from 'react';
import { Map, GoogleApiWrapper, InfoWindow, Marker, Polyline } from 'google-maps-react';
import Loader from 'react-loader-spinner'

const LoadingContainer = (props) => (
  <div className="loaderItem">
    <Loader
      type="ThreeDots"
      color="#6497b1"
      height="100"	
      width="100"
    />
  </div>
)

const mapStyles = {
  width: '100%',
  height: '100%'
};

export const MapContainer = (props) => {
  const {
    users,
    onMouseoverMarker,
    onMouseoutMarker,
    activeMarker,
    showingInfoWindow,
    perUserDetails,
    google
  } = props;
  const renderMarker = () => {
    return users.map((user) => {
      return (<Marker
        key={user.id}
        onMouseover={onMouseoverMarker}
        position={user.address.geo}
        onMouseout={onMouseoutMarker}
      />)
    })
  }
  const renderInfoWindow = () => {
    return (<InfoWindow
      marker={activeMarker}
      visible={showingInfoWindow}
      >
      <div className="info-window-text">
        <h2>{perUserDetails.name}</h2>
        Below is {perUserDetails.name} peronal information.
        <p>Company Name: {perUserDetails.company.name}</p>
        <p>Address: {`${perUserDetails.address.suite}, ${perUserDetails.address.street} ${perUserDetails.address.city}.`}</p>
        <p>Website: {perUserDetails.website}</p>
      </div>
    </InfoWindow>)
  }

  return (
      <Map
        google={google}
        zoom={2.2}
        className={'map'}
        style={mapStyles}
      >
      {renderMarker()}
      {renderInfoWindow()}
      </Map>
  )
}

MapContainer.defaultProps = {
 users: []
};

export default GoogleApiWrapper({
  apiKey: '',
  LoadingContainer
})(MapContainer);