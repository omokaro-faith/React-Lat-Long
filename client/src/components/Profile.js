import React from 'react';

const Profile = () => (
  <p>This is your Profile page</p>
)

export default Profile;