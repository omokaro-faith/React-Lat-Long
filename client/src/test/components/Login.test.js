import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../../components/Login';
import { start } from 'repl';

test('should render the Login page', () => {
  const wrapper = shallow(<Login />);
  expect(wrapper).toMatchSnapshot();
});

test('should call Login button on login click', (done) => {
  const startLogin = jest.fn();
  const wrapper = shallow(<Login startLogin={startLogin}/>);

  wrapper.find('button').simulate('click');
  expect(startLogin).toHaveBeenCalled();
  done();
});
