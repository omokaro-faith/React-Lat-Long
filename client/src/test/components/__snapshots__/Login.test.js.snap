// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Login page 1`] = `
<div
  className="container"
>
  <div>
    <img
      height="100%"
      id="container__image"
      src="btn-picture.jpg"
      width="100%"
    />
    <button
      className="button is-medium"
      onClick={[Function]}
    >
      <span>
        Login With Google
      </span>
    </button>
  </div>
</div>
`;
